frontend/src/pages/collaboration-service/CollaborationPage.tsx

```diff
@@ -1,22 +1,99 @@
 import { Link } from "react-router-dom";
 import MainContainer from "@/components/common/MainContainer";
 import PageHeader from "@/components/common/PageHeader";
-import { useAuth } from "@/contexts/AuthContext";
-import { useState } from 'react'
+import { useState, useEffect } from 'react'
 import { ChevronDown } from 'lucide-react'
 import { Button } from "@/components/ui/button"
 import TextEditor from "@/components/collaboration-service/TextEditor";
-import { useParams } from "react-router-dom"
+import { useNavigate } from "react-router-dom"
+import { getUserById } from "@/api/user-service/UserService";
+import { User } from "@/api/user-service/User";
+import { Question } from "@/api/question-service/Question";
+import { fetchQuestion } from "@/api/question-service/QuestionService";
+import { useAuth } from "@/contexts/AuthContext";
+import { removeUserFromCollabStore, isUserInCollabStore, getCollabInfo } from "@/api/collaboration-service/CollaborationService";
 
 export default function CollaborationPage() {
-  const { documentId } = useParams()
-  console.log(`documentId is: ${documentId}`)
+  const [roomId, setRoomId] = useState<string | null>(null)
+  const [matchedUserId, setMatchedUserId] = useState<string | null>(null)
+  const [questionId, setQuestionId] = useState<string | null>(null)
+
+  const { auth } = useAuth()
+  const navigate = useNavigate();
+
+  const [matchedUser, setMatchedUser] = useState<User | null>(null)
+  const [question, setQuestion] = useState<Question | null>(null)
+  
+  // When user enters this page, check if his ID is in collabStore. If isn't, block the user from entering this page
+  useEffect(() => {
+    const checkIfUserInStore = async () => {
+      if (auth.id === null) return
+
+      try {
+        const response1 = await isUserInCollabStore(auth.id)
+        if (response1.status === 200) {
+          // Using the user's ID, retrieve collaboration details
+          console.log('getting collab info..')
+          const response2 = await getCollabInfo(auth.id)
+          const data = response2.data
+  
+          setRoomId(data.roomId)
+          setMatchedUserId(data.matchedUserId)
+          setQuestionId(data.questionId)
+        } else {
+          // Means that user is not in user store, so he cannot access the collab-page
+          navigate("/matching/start")
+        }
+      } catch (error) {
+        console.error(error)
+      }
+    }
+
+    checkIfUserInStore()
+  }, [])
```

Comment:
(Continue from 1) 2. I am thinking of storing the question information in the response of collab store then render in frontend, instead of sending another request from frontend to question service. What do u think? (Feel free to comment for the rest!)